crypto/ocsp/ocsp_print.c

```diff
@@ -62,3 +78,190 @@ const char *OCSP_crl_reason_str(long s) {
   size_t tbl_size = (sizeof(reason_tbl) / sizeof((reason_tbl)[0]));
   return do_table2string(s, reason_tbl, tbl_size);
 }
+
+int OCSP_REQUEST_print(BIO *bp, OCSP_REQUEST *req, unsigned long flags) {
+  long l;
+  OCSP_CERTID *cid = NULL;
+  OCSP_ONEREQ *one = NULL;
+  OCSP_REQINFO *inf = req->tbsRequest;
+  OCSP_SIGNATURE *sig = req->optionalSignature;
+
+  if (BIO_write(bp, "OCSP Request Data:\n", 19) <= 0) {
```

Comment:
What is the difference between BIO_write, BIO_printf, and BIO_puts? Why do you alternate between them in this function?